app/grandchallenge/codebuild/tasks.py

```diff
@@ -0,0 +1,28 @@
+from celery import shared_task
+from django.apps import apps
+from django.conf import settings
+
+from grandchallenge.algorithms.models import Algorithm
+from grandchallenge.codebuild.client import CodeBuildClient
+
+
+@shared_task()
+def create_algorithm_image(*, pk):
```

Comment:
All celery tasks need to be idempotent and (ideally) short lived. I think if this is called twice with the same arguments now then it will create a new job every time. Then, rather than blocking for completion add a new celery task that checks for completion using the countdown argument (eg https://github.com/comic/grand-challenge.org/blob/master/app/grandchallenge/workspaces/tasks.py#L84 and how that is used in a chain, note that MaxRetriesExceeded should be handled). We also need to handle setting a time limit on the build task.

Reply:
I'll look into that. I'm currently adding the `Build` model which should make this a bit easier as we don't need to save the state in the client anymore. We need to think of an acceptable time limit for the build to finish though, as the build might take a while.

Reply:
It's just a build right? The users are probably not compiling a huge amount and not running tests so I think that a 15 minute limit would be a good starting point.

Reply:
Sounds good to me. I think the average build time for my (very simple) algorithm was under a minute.